app/controllers/api/v1/trends/tags_controller.rb

```diff
@@ -8,6 +8,7 @@ class Api::V1::Trends::TagsController < Api::BaseController
   DEFAULT_TAGS_LIMIT = 10
 
   def index
+    cache_if_unauthenticated!
```

Comment:
This is very likely to cause inconsistent page cache, especially since trends pagination does not work with an underlying ID and depends on the rank of the trend items.